### ISSUE TITLE
CloudFilesProvider.CreateObject doesn't support streams of unknown length

### ISSUE THREAD

**Issue by sharwell** [openstacknetsdk/openstack.net#211]
The current implementation of `CreateObject` requires the stream implement the `Length` property. It would be helpful to also support streams where this property throws a `NotSupportedException`.

**Comment by LightCZ** [comment]
this is actually pretty usefull feature, because when you proxy the stream into Swift you sometime dont know proper lenght of the stream.

**Comment by JoshMcCullough** [comment]
Agreed. I shouldn't have to create an intermediary stream to work around this issue.
